go/nbs/store.go

```diff
@@ -405,6 +405,18 @@ func (nbs *NomsBlockStore) updateManifest(current, last hash.Hash) error {
 		return errLastRootMismatch
 	}
 
+	if upstream, doomed := nbs.mm.updateWillFail(nbs.manifestLock); doomed {
+		// Pre-emptive optimistic lock failure. Someone else in-process moved to the root, the set of tables, or both out from under us.
```

Comment:
maybe lift this into a local `handleOptimisticLockFailure := func() {...}` that can be used below as well?